tiledb/tests/test_libtiledb.py

```diff
@@ -3477,6 +3477,16 @@ def test_from_numpy_timestamp(self):
         with tiledb.open(path, timestamp=(10, 10)) as A:
             assert A.nonempty_domain() == ((0, 2),)
 
+    @pytest.mark.parametrize("empty_str", ["", b""])
+    @pytest.mark.parametrize("num_strs", [1, 1000])
+    def test_from_numpy_empty_str(self, empty_str, num_strs):
+        uri = self.path("test_from_numpy_empty_str")
+        np_array = np.asarray([empty_str] * num_strs, dtype="O")
+        tiledb.from_numpy(uri, np_array)
+
+        with tiledb.open(uri, "r") as A:
+            assert_array_equal(A[:], np_array)
```

Comment:
Should test the `.df` reader here as well if pandas is available.

Reply:
Using Arrow and then converting to df is giving the NULL character instead of an empty string. .df works fine when not using Arrow.